app/jobs/submissions_job.rb

```diff
@@ -23,11 +23,16 @@ def perform(groupings, options = {})
     groupings.each do |grouping|
       m_logger.log("Now collecting: #{assignment.short_identifier} for grouping: " +
                    grouping.id.to_s)
-      if options[:revision_identifier].nil?
+      if !assignment.scanned_exam? && options[:revision_identifier].nil?
         time = options[:collection_dates]&.fetch(grouping.id, nil) || grouping.collection_date
         new_submission = Submission.create_by_timestamp(grouping, time)
       else
-        new_submission = Submission.create_by_revision_identifier(grouping, options[:revision_identifier])
+        revision_id = if assignment.scanned_exam? && options[:revision_identifier].nil?
```

Comment:
I feel like this is more complicated than it needs to be. I liked what you described originally at our meeting about using timestamps; in the "if block" above, how about setting `time = Time.current` if the assignment is a scanned exam?

Reply:
Yeah that sounds good. This was my initial approach but as mentioned I ran into bugs with testing. After changing it now though, those bugs are gone (I think I made errors with my original tests that I have since fixed).